core/embed/rust/src/ui/screens.rs

```diff
@@ -8,33 +8,26 @@ pub use super::model_tt::screens::*;
 use crate::ui::util::from_c_str;
 
 #[no_mangle]
-extern "C" fn screen_fatal_error_c(msg: *const cty::c_char, file: *const cty::c_char) {
-    let msg = if msg.is_null() {
-        None
-    } else {
-        unsafe { from_c_str(msg) }
-    };
-    let file = if file.is_null() {
+extern "C" fn screen_fatal_error_r(
+    title: *const cty::c_char,
+    msg: *const cty::c_char,
+    footer: *const cty::c_char,
+) {
+    let title = if title.is_null() {
```

Comment:
We use this construct three times in a row here, would it be worth to create a macro out of this?

```rs
if XXX.is_null() {
    ""
} else {
    unwrap!(unsafe { from_c_str(XXX) })
}
```

Reply:
https://github.com/trezor/trezor-firmware/commit/72eb69847e26b134df18f21ad7c7169b0a7c74c8